src/isar/state_machine/states/idle.py

```diff
@@ -37,4 +48,38 @@ def _run(self) -> None:
                 break
             time.sleep(self.state_machine.sleep_time)
 
+            time_from_last_robot_status_poll = (
+                time.time() - self.last_robot_status_poll_time
+            )
+            if (
+                time_from_last_robot_status_poll
+                < settings.ROBOT_API_STATUS_POLL_INTERVAL
+            ):
+                continue
+
+            if not self.robot_status_thread:
+                self.robot_status_thread = ThreadedRequest(
+                    request_func=self.state_machine.robot.robot_status
+                )
+                self.robot_status_thread.start_thread(
+                    name="State Machine Offline Get Robot Status"
+                )
+
+            try:
+                robot_status: RobotStatus = self.robot_status_thread.get_output()
+            except ThreadedRequestNotFinishedError:
+                time.sleep(self.state_machine.sleep_time)
+                continue
+
+            except (RobotException,) as e:
+                self.logger.error(
+                    f"Failed to get robot status because: {e.error_description}"
+                )
```

Comment:
Could we maybe move this to another function in order to shorten the function and make it a bit easier to read?

Reply:
We can standardise this in another PR